File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/Sudoku/Sudoku_VisualBasic/Controls/NoFlickerPanel.vb
'--------------------------------------------------------------------------
' 
'  Copyright (c) Microsoft Corporation.  All rights reserved. 
' 
'  File: NoFlickerPanel.vb
'
'  Description: A double-buffering and redraw-on-resizing panel.
' 
'--------------------------------------------------------------------------


Namespace Microsoft.ParallelComputingPlatform.ParallelExtensions.Samples.Sudoku.Controls
	''' <summary>A derived panel that uses double buffering to prevent flicker.</summary>
	<ToolboxBitmap(GetType(Panel))>
	Friend Class NoFlickerPanel
		Inherits Panel
		''' <summary>Initializes the panel.</summary>
		Public Sub New()
            SetStyle(ControlStyles.DoubleBuffer Or ControlStyles.AllPaintingInWmPaint Or
                     ControlStyles.UserPaint Or ControlStyles.ResizeRedraw Or ControlStyles.SupportsTransparentBackColor, True)
		End Sub
	End Class
End Namespace

File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/Sudoku/Sudoku_CSharp/Controls/NoFlickerPanel.cs
//--------------------------------------------------------------------------
// 
//  Copyright (c) Microsoft Corporation.  All rights reserved. 
// 
//  File: NoFlickerPanel.cs
//
//  Description: A double-buffering and redraw-on-resizing panel.
// 
//--------------------------------------------------------------------------

using System.Drawing;
using System.Windows.Forms;

namespace Microsoft.ParallelComputingPlatform.ParallelExtensions.Samples.Sudoku.Controls
{
	/// <summary>A derived panel that uses double buffering to prevent flicker.</summary>
	[ToolboxBitmap(typeof(Panel))]
	internal class NoFlickerPanel : Panel
	{
		/// <summary>Initializes the panel.</summary>
		public NoFlickerPanel()
		{
			SetStyle(
				ControlStyles.DoubleBuffer | 
				ControlStyles.AllPaintingInWmPaint | 
				ControlStyles.UserPaint | 
				ControlStyles.ResizeRedraw | 
				ControlStyles.SupportsTransparentBackColor, true);
		}
	}
}